lib/internal/util/parse_args/parse_args.js

```diff
@@ -118,12 +128,22 @@ function checkOptionUsage(config, token) {
  * @param {string|undefined} optionValue - value from user args
  * @param {object} options - option configs, from parseArgs({ options })
  * @param {object} values - option values returned in `values` by parseArgs
+ * @param {boolean} allowNegative - allow negative optinons if true
  */
-function storeOption(longOption, optionValue, options, values) {
+function storeOption(longOption, optionValue, options, values, allowNegative) {
   if (longOption === '__proto__') {
     return; // No. Just no.
   }
 
+  if (allowNegative && StringPrototypeStartsWith(longOption, 'no-')) {
```

Comment:
Suggested change for edge case which only occurs when `strict:false` so that do not look for negation if user specified value like `--no-foo=xyz`.
```suggestion
  if (allowNegative && StringPrototypeStartsWith(longOption, 'no-') && optionValue == undefined) {
```

Reply:
LGTM